File: app/src/main/java/com/example/notes/DatePickerFragment.java
package com.example.notes;


import static com.example.notes.NotesFragment.SELECTED_NOTE;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.DatePicker;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;


public class DatePickerFragment extends Fragment {

    public DatePickerFragment() {
        // Required empty public constructor
    }

    public static DatePickerFragment newInstance(Note note) {
        DatePickerFragment fragment = new DatePickerFragment();
        Bundle args = new Bundle();
        args.putParcelable(SELECTED_NOTE, note);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_date_picker, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Bundle arguments = getArguments();
        System.out.println(arguments + "date");
        if (arguments != null) {
            Note note = arguments.getParcelable(SELECTED_NOTE);
            DatePicker datePicker = view.findViewById(R.id.datePicker);
            datePicker.init(note.date[2], note.date[1] - 1, note.date[0], new DatePicker.OnDateChangedListener() {

                @Override
                public void onDateChanged(DatePicker view_date, int year, int monthOfYear, int dayOfMonth) {
                    note.date[0] = view_date.getDayOfMonth();
                    note.date[1] = view_date.getMonth() + 1;
                    note.date[2] = view_date.getYear();
                }
            });
        }
    }
}

File: app/src/main/java/com/example/notes/DatePickerFragment.kt
package com.example.notes

import android.content.res.Configuration
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import com.example.notes.R
import com.example.notes.ParentDateTimeFragment
import android.widget.DatePicker
import android.widget.DatePicker.OnDateChangedListener
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.example.notes.DatePickerFragment

class DatePickerFragment : Fragment() {
    private var layout: View? = null
    private var inflator: LayoutInflater? = null
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_date_picker, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val arguments = arguments
        // System.out.println(arguments + "date");
        if (arguments != null) {
            val note = arguments.getParcelable<Note>(ParentDateTimeFragment.SELECTED_NOTE)
            val datePicker = view.findViewById<DatePicker>(R.id.datePicker)
            if (!isLandscape) {
                datePicker.spinnersShown = false
                datePicker.calendarViewShown = true
            }
            datePicker.init(
                note!!.date[2],
                note.date[1] - 1,
                note.date[0]
            ) { view_date, year, monthOfYear, dayOfMonth ->
                note.date[0] = view_date.dayOfMonth
                note.date[1] = view_date.month + 1
                note.date[2] = view_date.year
                val notesd = Note.notes
                val indexd = notesd.indexOf(note)
                Note.notes[indexd] = note
                inflator = layoutInflater
                layout = inflator!!.inflate(
                    R.layout.fragment_note,
                    activity!!.findViewById(R.id.linear_layout_note)
                )
                val time_date_alarm = layout.findViewById<TextView>(R.id.time_date_alarm_view)
                time_date_alarm.text = note.timeDateAlarm
            }
        }
    }

    private val isLandscape: Boolean
        private get() = (resources.configuration.orientation
                == Configuration.ORIENTATION_LANDSCAPE)

    companion object {
        @JvmStatic
        fun newInstance(note: Note?): DatePickerFragment {
            val fragment = DatePickerFragment()
            val args = Bundle()
            args.putParcelable(ParentDateTimeFragment.SELECTED_NOTE, note)
            fragment.arguments = args
            return fragment
        }
    }
}